### ISSUE TITLE
Add new MouseFilter to Control node

### ISSUE THREAD

**Issue by ghsoares** [godotengine/godot-proposals#1322]
**Describe the project you are working on:**
An OS Simulator game

**Describe the problem or limitation you are having in your project:**
I'm trying to create a custom window control node where there's a _gui_input function to handle resizing, the problem is that the children is obscuring the input because of the MouseFilter, but I can't disable it. The solution would be add a control child of the window control to get any mouse event, and I though that the MouseFilter.Pass would report the input and continue the propagation, but after testing and reading the docs, the behaviour is that the input stops if is handled.

I need a MouseFilter constant that allows me to handle input event, without stoping propagation (_input is not an option because I need to handle the input inside the window, and still stop the propagation to the windows behind it).

I'm not using WindowDialog to create the UI because the Popup node only shows up as modal form and overlays other nodes even if in the Scene Tree is behind (even calling show() manually).

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
The feature would be a MouseFilter constant that allows the control node to handle the input and continue event propagation, if a control behind it have a _unhandled_input function, it fires normally.

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
I don't know how the MouseFilter works in code, but I think that when it reaches a control that handles input and have this MouseFilter constant, it don't call a function to stop the propagation.

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
No, bacause it needs to change the core code. Or building a custom Godot version that add this feature.

**Is there a reason why this should be core and not an add-on in the asset library?:**
Because as I said, I think that no script can change input behaviour beyound the *_input methods provided.

**Comment by KoBeWi** [comment]
I think this can be worked around by using Area2Ds, although mixing them with Controls is a rather hacky solution.

**Comment by ghsoares** [comment]
@KoBeWi yes, is a very counterintuitive aproach, because transforms between Control and Node2D aren't very supported, so I would need to manually from script change the Area2D CollisionShape's dimensions to match the window Control node Rect2.
